### PR TITLE
10860 sync information in node model

### LINKED ISSUE #10860: Avoid request to get tableModel from source nodes

### PR DESCRIPTION
**Pull request by javitonino**
Fixes #10860

### CODE REVIEW

File: lib/assets/javascripts/cartodb3/data/analysis-definition-nodes-collection.js [hunk 1]
```diff
@@ -12,8 +12,14 @@ module.exports = Backbone.Collection.extend({
 
   initialize: function (models, opts) {
     if (!opts.configModel) throw new Error('configModel is required');
+    if (!opts.relatedTableModels) throw new Error('relatedTableModels is required');
```
Comment (javitonino) [inline lib/assets/javascripts/cartodb3/data/analysis-definition-nodes-collection.js]: Should this be a collection? Right now it's an array.

File: lib/assets/javascripts/cartodb3/data/analysis-definition-nodes-collection.js [hunk 2]
```diff
@@ -14,6 +14,11 @@ module.exports = Backbone.Collection.extend({
     if (!opts.configModel) throw new Error('configModel is required');
 
     this._configModel = opts.configModel;
+    this._relatedTableModels = opts.relatedTableModels || [];
```
Comment (javitonino) [inline lib/assets/javascripts/cartodb3/data/analysis-definition-nodes-collection.js]: This is an array, but maybe it should be a collection?

**Comment by javitonino** [comment]
Ready for a first CR @xavijam.

**Comment by xavijam** [comment]
In my opinion, we should try to pass the data, not the whole model to the analysis source node model. Same logic we follow for layers collection. What do you think?

**Comment by javitonino** [comment]
Changes ready!

File: lib/assets/javascripts/cartodb3/data/analysis-definition-nodes-collection.js [hunk 3]
```diff
@@ -27,9 +32,17 @@ module.exports = Backbone.Collection.extend({
       opts
     );
 
-    return r.type === 'source'
-      ? new AnalysisDefinitionNodeSourceModel(r, opts)
-      : new AnalysisDefinitionNodeModel(r, opts);
+    if (r.type === 'source') {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/data/analysis-definition-nodes-collection.js]: Could we add a test in order to check this please?

**Comment by javitonino** [comment]
Added some tests.

**Pull request merged**